src/tpm2-setup/tpm2-setup.c

```diff
@@ -289,6 +291,11 @@ static int run(int argc, char *argv[]) {
         }
 
         r = load_public_key_tpm2(&tpm2_key);
+        if (ERRNO_IS_NEG_PRIVILEGE(r)) {
+                log_info("TPM is locked, cannot generate SRK.");
+                return 76; /* Special return value which means "TPM is locked, so not doing anything". This
```

Comment:
Please switch to DEFINE_MAIN_FUNCTION_WITH_POSITIVE_FAILURE, otherwise the exit code is discarded.